File: backend/api/models/Mihomo.go
package models

type Mihomo struct {
	Mode  string `json:"mode"`
	Proxy bool   `json:"proxy"`
	Tun   bool   `json:"tun"`

	Port        int    `json:"port"`
	BindAddress string `json:"bindAddress"`
	Stack       string `json:"stack"`
	Dns         bool   `json:"dns"`
	Ipv6        bool   `json:"ipv6"`
}

File: src/util/mihomo.ts
export function pUpdateMihomo(menuStore: any, settingStore: any, api: any): void {
    api.updateMihomo({
        mode: menuStore.rule,
        proxy: menuStore.proxy,
        tun: menuStore.tun,

        port: Number(settingStore.port),
        bindAddress: settingStore.bindAddress,
        stack: settingStore.stack,
        dns: settingStore.dns,
        ipv6: settingStore.ipv6,
    })
}